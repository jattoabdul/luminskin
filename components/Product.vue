<template>
  <div class="product thumbnail">
    <img :src="product.image_url" :alt="product.title">
    <div class="caption">
      <h3>{{ product.title }}</h3>
      <div class="product__price">
        {{ product.price | currency({symbol: getCurrencySymbol(currentCurrency.toLowerCase())}) }}
      </div>
      <div class="product__button-wrap">
        <button
          @click.stop="addToCart(product)"
          class="btn btn-primary"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Product',
  props: {
    product: {
      type: Object,
      default() {
        return {}
      }
    },
    currentCurrency: {
      type: String,
      default() {
        return 'USD'
      }
    }
  },
  computed: {},
  methods: {
    addToCart(product) {
      this.$eventBus.$emit('add-to-cart', { product })
    }
  }
}
</script>

<style scoped lang="scss">
.product {
  overflow: hidden;
}
.product__price {
  display: inline-block;
  line-height: 34px;
}
.product__button-wrap {
  float: right;
}
</style>
