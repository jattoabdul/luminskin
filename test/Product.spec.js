import { shallowMount } from '@vue/test-utils'
import Product from '@/components/Product.vue'
import Vue from 'vue'
import VueCurrencyFilter from 'vue-currency-filter'
import getSymbolFromCurrency from 'currency-symbol-map'

Vue.use(VueCurrencyFilter, {
  thousandsSeparator: ',',
  fractionCount: 2,
  fractionSeparator: '.',
  symbolPosition: 'front',
  symbolSpacing: false
})

Vue.mixin({
  methods: {
    getCurrencySymbol(currency) {
      const foundCurrency = getSymbolFromCurrency(currency)
      if (foundCurrency) {
        return foundCurrency
      }
      return currency
    }
  }
})

const product = {
  id: 1,
  title: 'product title',
  price: 100,
  image_url: 'image.svg'
}

const factory = () => {
  return shallowMount(Product, {
    propsData: {
      currentCurrency: 'USD',
      product
    }
  })
}

describe('Product', () => {
  test('mounts properly', () => {
    const wrapper = factory()
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
