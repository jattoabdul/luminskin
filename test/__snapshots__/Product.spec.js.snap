// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div class="product thumbnail"><img src="image.svg" alt="product title" class="product__image">
  <div class="caption">
    <h3 class="product__title">
      product title
    </h3>
    <div class="product__price">
      $100.00
    </div>
    <div class="product__button-wrap"><button class="btn btn-primary">
        Add to cart
      </button></div>
  </div>
</div>
`;
